.github/workflows/test.yml

```diff
@@ -446,10 +457,25 @@ jobs:
             gcloud compute instances delete "${INSTANCE}" --zone "${{ env.ZONE }}" --delete-disks all --quiet
           fi
 
-      - name: Get disk state name from gcloud
-        id: get-disk-name
+      # If a triggering file for a rebuild was changed on a PR or we manually triggered a rebuild,
+      # use the latest available disk state as the regenerate-stateful-disks job was executed
+      - name: Get latest state disk name from gcloud
+        id: get-latest-disk-name
+        if: ${{ (needs.build.outputs.any_changed == 'true' && github.event_name == 'pull_request') || github.event.inputs.regenerate-disks == 'true' }}
+        run: |
+          GCP_STATE_DISK=$(gcloud compute images list --filter="name~zebrad-cache-${{ env.GITHUB_REF_SLUG_URL }} AND name~-checkpoint" --format="value(NAME)" --sort-by=~creationTimestamp --limit=1)
+          echo "Disk: $GCP_STATE_DISK"
+          echo "Description: $(gcloud compute images describe $GCP_STATE_DISK --format='value(DESCRIPTION)')"
```

Comment:
Optional Lint: you could fix this shellcheck issue by splitting the subshell out, and assigning the result to its own variable.

This might also make the code easier to read.

Reply:
I'll do a full shellcheck fix PR, so this one doesn't get messy or implement a different approach from other files